File: tasks/1/B/speed.py
import sys

def solve():
    # Leemos todo el input de una vez para manejar los casos múltiples fácilmente
    input_data = sys.stdin.read().split()
    
    if not input_data:
        return

    iterator = iter(input_data)
    
    try:
        while True:
            # Intentamos leer n y t para el siguiente caso
            try:
                n_str = next(iterator)
                t_str = next(iterator)
            except StopIteration:
                break
            
            n = int(n_str)
            t = float(t_str)
            
            dists = []
            speeds = []
            min_speed = float('inf')
            
            # Leemos los n segmentos
            for _ in range(n):
                d = float(next(iterator))
                s = float(next(iterator))
                dists.append(d)
                speeds.append(s)
                if s < min_speed:
                    min_speed = s
            
            # Configuración de la Bisección
            # low: Debe ser mayor que -min_speed para evitar división por cero.
            # Se añade un epsilon pequeño para asegurar que el denominador sea positivo.
            low = -min_speed + 1e-9
            high = 10000000.0  # Un valor suficientemente grande
            
            # Usamos 100 iteraciones fijas.
            # Esto es estándar en Python para problemas de geometría/búsqueda binaria
            # para evitar problemas con epsilons y bucles infinitos.
            for _ in range(100):
                mid = (low + high) / 2
                
                # Calculamos el tiempo total con la constante 'mid'
                curr_t = 0.0
                for i in range(n):
                    curr_t += dists[i] / (speeds[i] + mid)
                
                # Lógica de decisión:
                # Si el tiempo calculado es mayor que el objetivo (t), vamos muy lento.
                # Necesitamos aumentar la velocidad, por lo tanto aumentamos c (mid).
                # Movemos el límite inferior hacia arriba.
                if curr_t > t:
                    low = mid
                else:
                    high = mid
            
            # Imprimimos el resultado con 4 decimales como pide el problema
            print(f"{mid:.9f}")
            
    except StopIteration:
        pass

if __name__ == '__main__':
    solve()

File: tasks/1/B/speed.cpp
/*
Jose David Ruano Burbano  8982982
Tarea 1 - Need for Speed
ADA - Análisis y Diseño de Algoritmos 2026-1 

Análisis de complejidad:
La complejidad temporal de este algoritmo es O(N * log(R/eps)), donde N es el número de segmentos, 
R es el rango de búsqueda inicial y eps es la precisión requerida.
En cada iteración de la bisección recorremos los N segmentos para calcular el tiempo total.
*/

#include <iostream>
#include <vector>
#include <cstdio>
#include <algorithm>

using namespace std;

int n;
double t;
vector<double> dists;
vector<double> speeds;

// Funcion que calcula el tiempo total del viaje dado un valor de constante c
double calculate_time(double c) {
    double total_time = 0;
    for (int i = 0; i < n; ++i) {
        total_time += dists[i] / (speeds[i] + c);
    }
    return total_time;
}

int main() {
    while (cin >> n >> t) {
        dists.resize(n);
        speeds.resize(n);
        double min_speed = 1001.0; // Valor inicial mayor al maximo posible (1000)

        for (int i = 0; i < n; ++i) {
            cin >> dists[i] >> speeds[i];
            // Buscamos la velocidad de lectura mínima para definir el límite inferior
            if (speeds[i] < min_speed) {
                min_speed = speeds[i];
            }
        }

        // Definición de límites para la biseccion
        // low: debe ser mayor que -min_speed para evitar division por cero o tiempos negativos.
        // high: un valor suficientemente grande donde el tiempo sea muy pequeño.
        double low = -min_speed + 1e-9; 
        double high = 10000000.0; 
        double mid;
        const double eps = 1e-9; 
        
        while (high - low > eps) {
            mid = (low + high) / 2;
            // Si el tiempo con mid es mayor al tiempo objetivo 't',
            // significa que vamos lento. Necesitamos aumentar la velocidad (es decir aumentar c).
            // esto ya que la función tiempo es decreciente respecto a c:
            if (calculate_time(mid) > t) {
                low = mid;
            } else {
                high = mid;
            }
        }

        printf("%.9f\n", mid);
    }

    return 0;
}